Tribler/Test/Core/CreditMining/test_creditmining_sys.py

```diff
@@ -0,0 +1,532 @@
+                """
+                function to emulate get_torrent in torrent_db
+                """
+                return {'C.torrent_id': 93, 'category': u'Compressed', 'torrent_id': 41,
+                        'infohash': src_obj.torrents.keys()[0], 'length': 1150844928, 'last_tracker_check': 10001,
+                        'myDownloadHistory': False, 'name': u'ubuntu-15.04-desktop-amd64.iso',
+                        'num_leechers': 999, 'num_seeders': 123, 'status': u'unknown', 'tracker_check_retries': 0}
+            self.boosting_manager.torrent_db.getTorrent = _get_tor_dummy
+            self.session.notifier.notify(NTFY_TORRENTS, NTFY_UPDATE, src_obj.torrents.keys()[0])
+
+            # log it
+            self.boosting_manager.log_statistics()
+
+            defer_param.callback(src)
+        return defer_param
+
+    def check_source(self, src, defer_param=None, ready=True):
+        """
+        function to check if a source is ready initializing
+        """
+        if defer_param is None:
+            defer_param = defer.Deferred()
+
+        src_obj = self.boosting_manager.get_source_object(src)
+
+        if not ready:
+            defer_param.callback(src)
+        elif not src_obj or not src_obj.ready:
+            reactor.callLater(1, self.check_source, src, defer_param)
+        else:
+            defer_param.callback(src)
+
+        return defer_param
+
+
+class TestBoostingManagerSysRSS(TestBoostingManagerSys):
+    """
+    testing class for RSS (dummy) source
+    """
+
+    def setUp(self, autoload_discovery=True):
+        super(TestBoostingManagerSysRSS, self).setUp()
+
+        files_path, self.file_server_port = prepare_xml_rss(self.session_base_dir, 'test_rss_cm.xml')
+
+        shutil.copyfile(TORRENT_FILE, os.path.join(files_path, 'ubuntu.torrent'))
+        self.setUpFileServer(self.file_server_port, self.session_base_dir)
+
+        self.is_error_fired = False
+        # now the rss should be at :
+        # http://localhost:port/test_rss_cm.xml
+        # which resides in sessiondir/http_torrent_files
+
+    def set_boosting_settings(self):
+        super(TestBoostingManagerSysRSS, self).set_boosting_settings()
+        self.bsettings.auto_start_source = False
+
+    def setUpFileServer(self, port, path):
+        resource = File(path)
+        resource.putChild("err503", ResourceFailClass())
+        factory = Site(resource)
+        self._logger.debug("Listen to port %s, factory %s", port, factory)
+        self.file_server = reactor.listenTCP(port, factory)
+
+    def check_feed_err(self, src, err_defer=None):
+        """
+        check whether a feed is producing error or not
+        """
+        if err_defer is None:
+            err_defer = defer.Deferred()
+
+        if self.is_error_fired:
+            err_defer.callback(True)
+        else:
+            reactor.callLater(1, self.check_feed_err, src, err_defer)
+
+        return err_defer
+
+    @deferred(timeout=20)
```

Comment:
Why can this test take up to 20 seconds to complete? What's it making it take so long?